Add slack announcement bot doc (#7440)
kubernetes/community: communication/contributor-comms/automation/announcement-guidelines.md

# K8s Announcement Bot User guide

The source code for this bot is present in https://github.com/kubernetes-sigs/slack-infra/tree/main/slack-post-message and it is deployed using https://github.com/kubernetes/k8s.io/tree/main/apps/slack-infra

## Prerequisites

> ### Permissions

Reviewer: ```suggestion
### Permissions
To use the bot, the user must be a member of the Slack 'sig-contributor-comms' group. Contact @contributor-comms to obtain the necessary permissions or to post something on your behalf.
```

Author: updated

Reach out to the 'sig-contribex-comms' channel in kubernetes slack for permissions or for posting something on your behalf.

> ### Adding bot to a channel

Reviewer: ```suggestion
### Permissions
To use the bot, the user must be a member of the Slack 'sig-contribex-comms' group. Contact @contributor-comms to request permissions or to post on your behalf.
```

> In order for the bot to send message to a channel, it needs to be part of the channel first and before you add it, you need to join the channel. Although we have it added in most of the channels, it still might be required to be added to a new channel. Follow the below steps for doing this

Reviewer: ```suggestion
To allow the bot to send messages to a channel, it must first be added to the channel. Before adding it, you need to make sure that the bot has already joined the channel. Although the bot is added to most of the channels, it might still be necessary to add it to a new channel. Follow the steps below to do so.
```

Author: done

1. Go to the 'More' option in the top left side and then go to 'Apps'

> ![](https://i.imgur.com/Dk26HwP.png)

Reviewer: we can have the assets within a folder in our GH repo, rather relying on 3rd parties.

Reviewer 2: Yes, please put it all in our repos.

Author: added in folders. PTAL,
